deflect/StreamSendWorker.cpp

```diff
@@ -151,44 +167,54 @@ Stream::Future StreamSendWorker::enqueueImage(const ImageWrapper& image,
                                               const bool finish)
 {
     if (_pendingFinish)
-        throw(std::runtime_error{"Pending finish, no send allowed"});
+    {
+        return make_exception_future<bool>(std::make_exception_ptr(
+            std::invalid_argument("Pending finish, no send allowed")));
+    }
 
     if (image.compressionPolicy != COMPRESSION_ON && image.pixelFormat != RGBA)
     {
-        std::cerr << "Currently, RAW images can only be sent in RGBA format. "
-                     "Other formats support remain to be implemented."
-                  << std::endl;
-        return make_ready_future(false);
+        return make_exception_future<bool>(
+            std::make_exception_ptr(std::invalid_argument(
+                "Currently, RAW images can only be sent in RGBA format. Other "
+                "formats support remain to be implemented.")));
     }
 
     if (image.compressionPolicy == COMPRESSION_ON)
     {
         if (image.compressionQuality < 1 || image.compressionQuality > 100)
         {
-            std::cerr
-                << "JPEG compression quality must be between 1 and 100, got "
+            std::stringstream msg;
+            msg << "JPEG compression quality must be between 1 and 100, got "
                 << image.compressionQuality << std::endl;
-            return make_ready_future(false);
+            return make_exception_future<bool>(
+                std::make_exception_ptr(std::invalid_argument(msg.str())));
```

Comment:
maybe we could have two, or an overload for Exception=std::exception_ptr, something like:
```
template <typename T>
std::future<T> make_exception_future(std::exception_ptr&& e)
{
    std::promise<T> promise;
    promise.set_exception(std::move(e));
    return promise.get_future();
}
template <typename T, typename Exception>
std::future<T> make_exception_future(Exception&& e)
{
    return make_exception_future(std::make_exception_ptr(std::move(e)));
}


```